flow/alerting/alerting.go

```diff
@@ -412,36 +410,46 @@ func (a *Alerter) LogNonFlowCritical(ctx context.Context, eventType telemetry.Ev
 }
 
 func (a *Alerter) LogNonFlowEvent(ctx context.Context, eventType telemetry.EventType, key string, message string, level telemetry.Level) {
-	a.sendTelemetryMessage(ctx, string(eventType)+":"+key, message, level)
+	logger := shared.LoggerFromCtx(ctx)
+	a.sendTelemetryMessage(ctx, logger, string(eventType)+":"+key, message, level)
 }
 
 func (a *Alerter) LogFlowError(ctx context.Context, flowName string, err error) {
-	logger := logger.LoggerFromCtx(ctx)
 	errorWithStack := fmt.Sprintf("%+v", err)
+	logger := shared.LoggerFromCtx(ctx)
 	logger.Error(err.Error(), slog.Any("stack", errorWithStack))
-	_, err = a.CatalogPool.Exec(ctx,
-		"INSERT INTO peerdb_stats.flow_errors(flow_name,error_message,error_type) VALUES($1,$2,$3)",
-		flowName, errorWithStack, "error")
-	if err != nil {
+	if _, err := a.CatalogPool.Exec(
+		ctx, "INSERT INTO peerdb_stats.flow_errors(flow_name,error_message,error_type) VALUES($1,$2,$3)",
+		flowName, errorWithStack, "error",
+	); err != nil {
 		logger.Warn("failed to insert flow error", slog.Any("error", err))
 		return
 	}
-	a.sendTelemetryMessage(ctx, flowName, errorWithStack, telemetry.ERROR)
+	var tags []string
+	if errors.Is(err, context.Canceled) {
+		tags = append(tags, "errCanceled")
+	}
+	if errors.Is(err, io.EOF) || errors.Is(err, io.ErrUnexpectedEOF) {
+		tags = append(tags, "errEOF")
```

Comment:
should we do "errType:errEOF" instead? might be more robust to parse for all the cases

Reply:
or at least `err:EOF`